kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualClusterModel.java

```diff
@@ -83,7 +83,10 @@ public VirtualClusterModel(String clusterName,
     }
 
     public void logVirtualClusterSummary() {
-        var upstreamHostPort = targetCluster.bootstrapServersList().get(0);
+        var upstreamBootstrapServers = targetCluster.bootstrapServersList();
+        var upstreamSummary = upstreamBootstrapServers.size() == 1 
+            ? upstreamBootstrapServers.get(0).toString()
```

Comment:
I'd suggest not special casing the size == 1 case, Just String.join() the list separating by comma.

Reply:
updated